include/CL/sycl/vendor/trisycl/scope/device/detail/device.hpp

```diff
@@ -0,0 +1,97 @@
+#ifndef TRISYCL_SYCL_VENDOR_TRISYCL_SCOPE_DEVICE_DETAIL_DEVICE_HPP
+#define TRISYCL_SYCL_VENDOR_TRISYCL_SCOPE_DEVICE_DETAIL_DEVICE_HPP
+
+/** \file Implementation details of the SYCL conceptual device with a
+    device scope storage
+
+    \todo The device and platform are very similar. Use a common
+    mix-in to implement both of them?
+
+    Ronan at Keryell point FR
+
+    This file is distributed under the University of Illinois Open Source
+    License. See LICENSE.TXT for details.
+*/
+
+#include "CL/sycl/device.hpp"
+
+namespace cl::sycl::vendor::trisycl::scope::detail {
+
+/** \addtogroup vendor_trisycl_scope triSYCL extension for
+    storage scopes
+    @{
+*/
+
+/** A conceptual device implementing some device-scoped storage and
+    based on a platform with platform-scoped storage
+*/
+template <typename DeviceStorage
+          , typename ScopedPlatform>
+class device {
+  /// The storage-less device behind the scene
+  cl::sycl::device d;
+
+  /** The device-scoped storage default-initialized
+
+      \todo For now it is allocated on the host in this CPU emulation
+      but a devils compiler and run time can create this on a real
```

Comment:
Possible todo comment mistake devils supposed to be device?

Reply:
Haha my spell checker seems to find some evil in the device details... :-)